src/Disciplina/DB/DSL/Interpret/SimpleTxDB.hs

```diff
@@ -1,92 +1,147 @@
-  return $ db ^..  traversed
-               . _SSTx
-               . filtered ((==sId).ciSubject._ptxCourseId)
+evalSimpleTxsQuery (SELECTTxs _ (TxHasSubjectId sId)) = do
+    db <- asks _getSimpleObj
+    subjToCourseMap <- asks _courseIdToSubjectId
+    let txs = db ^.. traversed
+                  . _SSTx
+                  . filtered (subjectIdHasCourseId subjToCourseMap . _ptxCourseId)
+    return txs
+  where subjectIdHasCourseId subjToCourseMap courseId =
+          case subjToCourseMap ^? at courseId . _Just . to (any (==sId)) of
+                 Just True -> True
+                 _         -> False
 
 -- | Evaluator for query: find Txs in db with grade == g
 evalSimpleTxsQuery (SELECTTxs _ (TxGradeEq grade)) = do
-  db <- ask
-  return $ db ^..  traversed
-               . _SSTx
-               . filtered ((== Just grade).getGrade._ptxPayload)
+    db <- asks _getSimpleObj
+    return $ db ^.. traversed
+                 . _SSTx
+                 . filtered ((== Just grade).getGrade._ptxPayload)
   where getGrade (EducatorTx (GradeCourse g)) = Just g
-        getGrade _ = Nothing
+        getGrade _                            = Nothing
 
 -- | Evaluator for query: find Txs in db with grade >= g
 evalSimpleTxsQuery (SELECTTxs _ (_ :>= grade)) = do
-  db <- ask
-  return $ db ^..  traversed
-               . _SSTx
-               . filtered ((>= Just grade).getGrade._ptxPayload)
+    db <- asks _getSimpleObj
+    return $ db ^.. traversed
+                 . _SSTx
+                 . filtered ((>= Just grade).getGrade._ptxPayload)
   where getGrade (EducatorTx (GradeCourse g)) = Just g
-        getGrade _ = Nothing
+        getGrade _                            = Nothing
 
 -- | Evaluator for AND query
 evalSimpleTxsQuery (SELECTTxs _ (a :& b)) =
-   intersect <$> runQuery (SELECTTxs WHERE a) <*> runQuery (SELECTTxs WHERE b)
+    intersect <$> runQuery (SELECTTxs WHERE a) <*> runQuery (SELECTTxs WHERE b)
 
 -- | Evaluator for OR query
 evalSimpleTxsQuery (SELECTTxs _ (a :|| b)) =
-   union <$> runQuery (SELECTTxs WHERE a) <*> runQuery (SELECTTxs WHERE b)
+    union <$> runQuery (SELECTTxs WHERE a) <*> runQuery (SELECTTxs WHERE b)
 
 -- | Evaluator for query: find all txs with subjectId which is descendant of sId
-evalSimpleTxsQuery (SELECTTxs _ (TxSubjectIsDescendantOf sId)) = do
-  db <- ask
-  return $ db ^..  traversed
-               . _SSTx
-               . filtered (isDescendantOf sId . ciSubject ._ptxCourseId)
-  where isDescendantOf x y = hasPathFromTo activityTypeGraphIndexed x y
+evalSimpleTxsQuery (SELECTTxs _ (TxHasDescendantOfSubjectId sId)) = do
+    db <- asks _getSimpleObj
+    subjToCourseMap <- asks _courseIdToSubjectId
+    let txs = db ^.. traversed
+                  . _SSTx
+                  . filtered (hasDescendantOf subjToCourseMap . _ptxCourseId)
+    return txs
+  where hasDescendantOf subjToCourseMap courseId =
+          case subjToCourseMap ^? at courseId . _Just . to (any (isDescendantOf sId)) of
```

Comment:
Same as in previous case. Also, could be done more efficiently by implementing a more general `hasPathFromTo` method which accepts a list of possible "to" vertices, but it's definitely not essential here.